File: Scripts/results/generated_code_results/Flutter/Shopping&E-commerce/Amazon/AccountScreen.dart
import 'package:flutter/material.dart';



class AccountScreen extends StatelessWidget {

  @override

  Widget build(BuildContext context) {

    return Scaffold(

      appBar: AppBar(

        leading: IconButton(

          icon: Icon(Icons.arrow_back),

          onPressed: () {

            Navigator.pushNamed(context, 'Profile');

          },

        ),

        title: TextField(

          decoration: InputDecoration(

            hintText: 'Search Amazon.co.uk',

            suffixIcon: Row(

              mainAxisSize: MainAxisSize.min,

              children: [

                Icon(Icons.camera_alt),

                Icon(Icons.mic),

              ],

            ),

          ),

        ),

      ),

      body: ListView(

        children: [

          Text('Orders', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),

          ListTile(

            title: Text('Your Orders'),

            trailing: Icon(Icons.chevron_right),

            onTap: () {

              Navigator.pushNamed(context, 'Orders');

            },

          ),

          ListTile(

            title: Text('Your Subscribe & Save'),

            trailing: Icon(Icons.chevron_right),

            onTap: () {

              // Navigate to Subscribe & Save page

            },

          ),

          ListTile(

            title: Text('Your Amazon Day'),

            trailing: Icon(Icons.chevron_right),

            onTap: () {

              // Navigate to Amazon Day page

            },

          ),

          Text('Payments', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),

          ListTile(

            title: Text('Payments and transactions'),

            trailing: Icon(Icons.chevron_right),

            onTap: () {

              // Navigate to Payments and transactions page

            },

          ),

          ListTile(

            title: Text('Manage gift card balance'),

            trailing: Icon(Icons.chevron_right),

            onTap: () {

              // Navigate to Manage gift card balance page

            },

          ),

          ListTile(

            title: Text('Shop with Points'),

            trailing: Icon(Icons.chevron_right),

            onTap: () {

              // Navigate to Shop with Points page

            },

          ),

          ListTile(

            title: Text('Top up account'),

            trailing: Icon(Icons.chevron_right),

            onTap: () {

              // Navigate to Top up account page

            },

          ),

          ListTile(

            title: Text('In-Store Promo Wallet'),

            trailing: Icon(Icons.chevron_right),

            onTap: () {

              // Navigate to In-Store Promo Wallet page

            },

          ),

          Text('Customer Service', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),

          ListTile(

            title: Text('Contact Us'),

            trailing: Icon(Icons.chevron_right),

            onTap: () {

              // Navigate to Contact Us page

            },

          ),

          Text('Account Settings', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),

          // Add more account settings options here

        ],

      ),

      bottomNavigationBar: BottomNavigationBar(

        type: BottomNavigationBarType.fixed,

        items: [

          BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),

          BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),

          BottomNavigationBarItem(icon: Icon(Icons.shopping_cart), label: 'Cart'),

          BottomNavigationBarItem(icon: Icon(Icons.menu), label: 'Menu'),

        ],

        onTap: (index) {

          if (index == 0) {

            Navigator.pushNamed(context, 'Home');

          } else if (index == 1) {

            Navigator.pushNamed(context, 'Profile');

          }

          // Handle other navigation items

        },

      ),

    );

  }

}

File: Scripts/results/generated_code_results/ablation_study/sam/Shopping&E-commerce/Amazon/AccountScreen.js
import React from 'react';

import { View, Text, StyleSheet, TouchableOpacity, ScrollView } from 'react-native';

import { useNavigation } from '@react-navigation/native';

import Icon from 'react-native-vector-icons/MaterialIcons';



const AccountScreen = () => {

  const navigation = useNavigation();



  const renderSection = (title, options) => (

    <View style={styles.section}>

      <Text style={styles.sectionTitle}>{title}</Text>

      {options.map((option, index) => (

        <TouchableOpacity

          key={index}

          style={styles.option}

          onPress={() => handleOptionPress(option)}

        >

          <Text style={styles.optionText}>{option}</Text>

          <Icon name="chevron-right" size={24} color="#888" />

        </TouchableOpacity>

      ))}

    </View>

  );



  const handleOptionPress = (option) => {

    switch (option) {

      case 'Your Orders':

        navigation.navigate('Orders');

        break;

      case 'Contact Us':

        // Handle Contact Us action

        break;

      default:

        // Handle other options

        break;

    }

  };



  return (

    <View style={styles.container}>

      <ScrollView>

        {renderSection('Orders', ['Your Orders', 'Your Subscribe & Save', 'Your Amazon Day'])}

        {renderSection('Payments', [

          'Payments and transactions',

          'Manage gift card balance',

          'Shop with Points',

          'Top up account',

          'In-Store Promo Wallet',

        ])}

        {renderSection('Customer Service', ['Contact Us'])}

        <View style={styles.section}>

          <Text style={styles.sectionTitle}>Account Settings</Text>

        </View>

      </ScrollView>

      <View style={styles.bottomBar}>

        <TouchableOpacity onPress={() => navigation.navigate('Home')}>

          <Icon name="home" size={24} color="#888" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Profile')}>

          <Icon name="person" size={24} color="#888" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Cart')}>

          <Icon name="shopping-cart" size={24} color="#888" />

        </TouchableOpacity>

        <TouchableOpacity>

          <Icon name="menu" size={24} color="#888" />

        </TouchableOpacity>

      </View>

    </View>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#f0f0f0',

  },

  section: {

    backgroundColor: '#fff',

    marginBottom: 10,

    paddingVertical: 10,

  },

  sectionTitle: {

    fontSize: 18,

    fontWeight: 'bold',

    paddingHorizontal: 15,

    marginBottom: 10,

  },

  option: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    alignItems: 'center',

    paddingVertical: 12,

    paddingHorizontal: 15,

  },

  optionText: {

    fontSize: 16,

  },

  bottomBar: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    alignItems: 'center',

    backgroundColor: '#fff',

    paddingVertical: 10,

    borderTopWidth: 1,

    borderTopColor: '#e0e0e0',

  },

});



export default AccountScreen;